libbeat/autodiscover/providers/kubernetes/config.go

```diff
@@ -102,6 +119,41 @@ func (c *Config) Validate() error {
 	if c.Scope != "node" && c.Scope != "cluster" {
 		return fmt.Errorf("invalid `scope` configured. supported values are `node` and `cluster`")
 	}
+
+	if c.Sharding.Count != 0 {
+		if c.Scope == "node" {
+			logp.L().Warnf("can not set sharding.count to `%d` when scope to `cluster`", c.Sharding.Count)
+			c.Sharding.Count = 0
+			c.Sharding.Instance = -1
+		} else {
+			// Ensure that if a user decides to run cluster mode sharded Beats, then each instance and its replica do leader election.
+			if c.Unique && c.LeaderLease != "" {
+				c.LeaderLease = fmt.Sprintf("%s-%d", c.LeaderLease, c.Sharding.Instance)
```

Comment:
Is this correct? `LeaderLease` has the meaning of defining the name of target lease between the different Beat candidates that try to gain the lock and be the leader. With this change each instance will create a different lease, no?